idea/src/org/jetbrains/kotlin/idea/inspections/RemoveRedundantConstructorParenthesesInspection.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.psi.*
+
+class RemoveRedundantConstructorParenthesesInspection : AbstractKotlinInspection() {
```

Comment:
I suspect it should be done other way. First of all, it's quite unclear why only constructors should be handled here, and why only case with one value parameter should be considered. And second, there is `MoveLambdaOutsideParenthesesIntention` which solves the same task. Probably the best we can do here is to convert it to the inspection using `AbstractApplicabilityBasedInspection`. I checked, and intention works in your case too.